src/platform/silabs/wifi/rs911x/platform/efx32_ncp_host.c

```diff
@@ -135,6 +107,45 @@ static void efx32_spi_init(void)
     GPIO_ExtIntConfig(INTERRUPT_PIN.port, INTERRUPT_PIN.pin, INTERRUPT_PIN.pin, true, false, true);
 }
 
+Ecode_t si91x_SPIDRV_MTransfer(SPIDRV_Handle_t handle, const void * txBuffer, void * rxBuffer, int count,
+                               SPIDRV_Callback_t callback)
+{
+    USART_TypeDef * usart = handle->initData.port;
+    uint8_t * tx          = (txBuffer != NULL) ? (uint8_t *) txBuffer : dummy_buffer;
+    uint8_t * rx          = (rxBuffer != NULL) ? (uint8_t *) rxBuffer : dummy_buffer;
+
+    if (count < 16)
+    {
+        while (count > 0)
+        {
+            while (!(usart->STATUS & USART_STATUS_TXBL))

```

Comment:
I vaguely remember this being discussed before. Why are we doing busy wait and manual byte send/received when count is < 16 ?

Reply:
DMA is not used in this case as it will not be the most efficient way to use SPI in case of small data size.